libs/usb-drive/src/usb_drive.ts

```diff
@@ -19,8 +20,8 @@ export type UsbDriveStatus =
 
 export interface UsbDrive {
   status(): Promise<UsbDriveStatus>;
-  eject(): Promise<void>;
-  format(): Promise<void>;
+  eject(loggingUserRole: LoggingUserRole): Promise<void>;
+  format(loggingUserRole: LoggingUserRole): Promise<void>;
```

Comment:
This API is kind of weird now. It makes me think that we're not separating concerns well when it comes to logging.

I wonder if we should initialize the logger with a way to query the current user role instead of passing it in when calling `log`...

E.g.:
```ts
const logger = new Logger({ auth });
logger.userLog(LogEventId.UserDidSomething); // queries auth for current user type
logger.systemLog(LogEventId.SystemDidSomething); // uses "system" user type
```

Reply:
It is weird. I think this is a good idea. There are lots of places where we're passing user roles around awkwardly to log with the role. It was a problem with the over-computation of tallies in the frontend. That was happening in an effect (bad start) that depended on the user role to log properly. Very silly.